lib/logstash/inputs/lumberjack.rb

```diff
@@ -126,4 +127,69 @@ def start_buffer_broker(output_queue)
       end
     end
   end
+
+  class Decoration < LogStash::Inputs::Base
+    public
+    def decorate(event)
+      super(event)
+    end
+  end # class Decoration
```

Comment:
@jordansissel I think we should decorate the events after they leave the plugins and do our best to merge the values?
Since decorating the event is mostly adding custom fields and tags we could generalize it outside of the plugin.